docs/proposals/DataTable/README.md

```diff
@@ -0,0 +1,165 @@
+# DataTable
+
+The `DataTable` Component would give the users options for pagination, sorting,
+and column visibility by leveraging some sort of API or service.
+
+## Design Patterns
+
+Tables are used to organize tabular information to help users understand and
+quickly scan for information.
+
+Some scenarios for Tables include the following:
+
+- Index Pages
+- Dashboards
+
+### Sorting
+
+Sorting can be enabled on column headers to improve usability.
+
+The default sorted column should be based on contextual based on user needs.
+
+To indicate the column being sorted, use an up (ascending) or down (descending)
+arrow next to the column title. Only 1 column can be sorted at a time. User can
+click on a sorted header to reverse the sorting order. Arrow will rotate 180
+degrees.
+
+Not all columns will need sorting enabled.
+
+# Column Visibility
+
+Not 100% sure if this is something to be implemented at component level but at
```

Comment:
it looks like we can pass this data to the table through state, though depending on the design around toggling visibility of table columns, this might be a separate component that gets this data and passes it to this table 🤔

Reply:
From this [example](https://tanstack.com/table/v8/docs/examples/react/column-visibility), it seems that column visibility can be accomplished using the library on the client side, what I'm not 100% sure is how to handle this when the pagination is done manually by leveraging an API.

Reply:
I agree with this approach, we will do some testing to make sure this is a working solution.

Reply:
It seems like you can just modify the state of the table to hide certain columns?